Enable system tablespace encryption again in 8.0.20

satya-bodapati opened the pull request:
1. Enable system tablespace encryption again in 8.0.20

2. Remove innodb.percona_sys_tablespace_encrypt_dblwr test as there is no doublewrite
   buffer in system tablespace anymore

percona-ysorokin commented on storage/innobase/fil/fil0fil.cc:
```diff
@@ -7748,6 +7748,18 @@ static bool fil_keyring_skip_encryption(const page_id_t &page_id) {
 @param[in]	space		table space */
 void fil_io_set_encryption(IORequest &req_type, const page_id_t &page_id,
                            fil_space_t *space) {
+  // TODO: now that dblwr doesn't exist in sys , should we encrypt all pages?
```
Original patch also used to have `req_type.is_encryption_disabled()` check here.
Any reason why it is not needed any more?

satya-bodapati replied:
I did a bad job not commenting on why I introduced this flag in IOrequest.

When system tablespace is encrypted, all pages in system tablespace dblwr would be written encrypted too. Page0 of every tablespace should be unencrypted. When such a page is written to system dblwr, it is written at doublewrite buffer offset. the page_id used is different. It is (0, offset_in_dblwr).  The check in fil_io_set_encryption() will fail to skip encryption for these pages in system dblwr.

Hence, for these special pages which are always supposed to be un-encrypted, we force them to be un-encrypted in system tablesapce dblwr.

SInce the dblwr in system tablespace is gone, this change is no longer required.

percona-ysorokin commented on storage/innobase/srv/srv0start.cc:
```diff
@@ -1996,6 +1996,58 @@ static lsn_t srv_prepare_to_delete_redo_log_files(ulint n_files) {
+static dberr_t srv_sys_enable_encryption(bool create_new_db) {
+  fil_space_t *space = fil_space_get(TRX_SYS_SPACE);
+  dberr_t err = DB_SUCCESS;
+
```
Original patch used to have
```
  // Fail startup if sys space is encrypted with crypt_data v1
  // This should only happen on upgrade
  if (srv_sys_space.keyring_encryption_info.page0_has_crypt_data &&
      srv_sys_space.keyring_encryption_info.type != CRYPT_SCHEME_UNENCRYPTED &&
      srv_sys_space.keyring_encryption_info.private_version == 1) {
    ib::error(ER_UPGRADE_KEYRING_V1_ENCRYPTION);
    return (DB_ERROR);
  }
```
here

satya-bodapati replied:
Hmm, this should be brought back again. If there is crypt_data old format in the system tablespace, we disallow upgrades. Thanks for finding this.

satya-bodapati replied:
Done!

satya-bodapati committed:
Bug#99935 : innodb_doublewrite_files is not correct when innodb_buffer_pool_size > 1G

Problem:
-------
According to documentation, default value of innodb_doublewrite_files is
innodb_buffer_pool_instances * 2

But when you start InnoDB with buffer pool size > 1G (the number of instances
is 8), it still creates 2 files only. Expected files is 16 (8*2)

Fix:
